src/pages/explore/wallets/_components/DynamicDownloadCard.tsx

```diff
@@ -0,0 +1,135 @@
+  keywords: {
+    mac?: string;
+    win?: string;
+    linux?: string;
+  };
+}
+
+interface DownloadLinks {
+  mac?: string;
+  win?: string;
+  linux?: string;
+}
+
+export function DynamicDownloadCard(
+  props: PropsWithChildren<DynamicDownloadCardProps>
+): JSX.Element {
+  const [downloadLinks, setDownloadLinks] = useState<DownloadLinks>();
+  const [isLoading, setLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    const links: DownloadLinks = {};
+
+    void getGitHubAssets(props.repoName).then((assets) => {
+      if (assets === undefined) {
+        setHasError(true);
+        return;
+      }
+
+      Object.keys(props.keywords).forEach((key) => {
+        const link = getLink(assets, props.keywords[key]);
+        if (link === undefined) {
+          setHasError(true);
+        }
+        links[key] = link;
+      });
+
+      setDownloadLinks(links);
+      setLoading(false);
+    });
+  }, []);
+
+  return (
+    <DownloadCard
+      title={props.title}
+      desc={props.desc}
+      imageSrc={props.imageSrc}
+      testid={props.testid}
+    >
+      <div className="flex items-center text-lg font-medium">
+        {isLoading || hasError ? (
+          <CardLink
+            descText="View on"
+            text={IconType.GITHUB}
+            url={`https://github.com/${props.repoName}/releases/latest`}
+          />
+        ) : (
+          <div className="flex items-center space-x-4 text-lg font-medium">
+            {props.keywords.mac !== undefined &&
+              downloadLinks?.mac !== undefined && (
+                <CardLink
+                  descText="Download for"
+                  text={IconType.MACOS}
+                  url={downloadLinks.mac}
+                />
+              )}
+            {props.keywords.win !== undefined &&
+              downloadLinks?.win !== undefined && (
+                <CardLink
+                  descText="Download for"
+                  text={IconType.WINDOWS}
+                  url={downloadLinks.win}
+                />
+              )}
+            {props.keywords.linux !== undefined &&
+              downloadLinks?.linux !== undefined && (
+                <CardLink
+                  descText="Download for"
+                  text={IconType.LINUX}
+                  url={downloadLinks.linux}
+                />
+              )}
+          </div>
+        )}
+      </div>
+    </DownloadCard>
+  );
+}
+
+export async function getGitHubAssets(
+  repoName: string
+): Promise<any | undefined> {
+  const baseUrl = `https://api.github.com/repos/${repoName}/releases/latest`;
+  const res = await fetch(baseUrl);
+  const json = await res.json();
+
+  const { assets } = json;
+
+  if (assets === undefined || assets.length === 0) {
+    return undefined;
+  }
+
+  return assets;
+}
+
+function getLink(assets, keyword: string): string | undefined {
+  const blockList = [".sha256", ".blockmap", ".yml"];
```

Comment:
i know this is copied from v1, but can we update the naming to not confused with block in blockchain?